sim/dex.ts

```diff
@@ -314,6 +314,23 @@ class ModdedDex {
 		}
 	}
 
+	getEffectivenessCode(
+		move: Move,
+		target: Pokemon
+	): number {
+		if (!this.getImmunity(move, target)) return 3;
+		if (move.ohko || move.damageCallback || move.damage) {
+			return 0;
+		}
+		const baseMod = this.getEffectiveness(move, target);
+		// @ts-ignore
```

Comment:
Can you make this `// @ts-ignore - dynamic lookup` or something similar? I'm trying to add explanations to when we use `// @ts-ignore` (like is [generally recommended](https://source.android.com/setup/contribute/code-style#use-standard-java-annotations) for `@SuppressWarnings` in Java), which helps make sure people are not ignoring things for the wrong reasons and makes it so we know which we can remove in the future when Typescript gets smarter (index signature crap) or we refactor.

Reply:
Sure!